auth/service.go

```diff
@@ -284,38 +291,43 @@ func (svc service) RemoveGroup(ctx context.Context, token, id string) error {
 	return svc.groups.Delete(ctx, id)
 }
 
-func (svc service) Unassign(ctx context.Context, token, memberID, groupID string) error {
+func (svc service) UpdateGroup(ctx context.Context, token string, group Group) (Group, error) {
 	if _, err := svc.Identify(ctx, token); err != nil {
-		return errors.Wrap(ErrUnauthorizedAccess, err)
+		return Group{}, errors.Wrap(ErrUnauthorizedAccess, err)
 	}
-	return svc.groups.Unassign(ctx, memberID, groupID)
+
+	group.UpdatedAt = getTimestmap()
+	return svc.groups.Update(ctx, group)
 }
 
-func (svc service) UpdateGroup(ctx context.Context, token string, g groups.Group) (groups.Group, error) {
+func (svc service) ViewGroup(ctx context.Context, token, id string) (Group, error) {
 	if _, err := svc.Identify(ctx, token); err != nil {
-		return groups.Group{}, errors.Wrap(ErrUnauthorizedAccess, err)
+		return Group{}, errors.Wrap(ErrUnauthorizedAccess, err)
 	}
-
-	return svc.groups.Update(ctx, g)
+	return svc.groups.RetrieveByID(ctx, id)
 }
 
-func (svc service) ViewGroup(ctx context.Context, token, id string) (groups.Group, error) {
+func (svc service) Assign(ctx context.Context, token string, groupID, groupType string, memberIDs ...string) error {
 	if _, err := svc.Identify(ctx, token); err != nil {
-		return groups.Group{}, errors.Wrap(ErrUnauthorizedAccess, err)
+		return errors.Wrap(ErrUnauthorizedAccess, err)
 	}
-	return svc.groups.RetrieveByID(ctx, id)
+	return svc.groups.Assign(ctx, groupID, groupType, memberIDs...)
 }
 
-func (svc service) Assign(ctx context.Context, token, memberID, groupID string) error {
+func (svc service) Unassign(ctx context.Context, token string, groupID string, memberIDs ...string) error {
 	if _, err := svc.Identify(ctx, token); err != nil {
 		return errors.Wrap(ErrUnauthorizedAccess, err)
 	}
-	return svc.groups.Assign(ctx, memberID, groupID)
+	return svc.groups.Unassign(ctx, groupID, memberIDs...)
 }
 
-func (svc service) ListMemberships(ctx context.Context, token string, memberID string, offset, limit uint64, gm groups.Metadata) (groups.GroupPage, error) {
+func (svc service) ListMemberships(ctx context.Context, token string, memberID string, pm PageMetadata) (GroupPage, error) {
 	if _, err := svc.Identify(ctx, token); err != nil {
-		return groups.GroupPage{}, errors.Wrap(ErrUnauthorizedAccess, err)
+		return GroupPage{}, errors.Wrap(ErrUnauthorizedAccess, err)
 	}
-	return svc.groups.Memberships(ctx, memberID, offset, limit, gm)
+	return svc.groups.Memberships(ctx, memberID, pm)
+}
+
+func getTimestmap() time.Time {
```

Comment:
This is used only twice in the code and does not seem to be particularly useful.

Reply:
someone may change rounding for example in one place and forgot to do it in other place